k9mail/src/main/java/com/fsck/k9/service/PollService.java

```diff
@@ -109,11 +109,6 @@ public void checkMailStarted(Context context, Account account) {
         }
 
         @Override
-        public void checkMailFailed(Context context, Account account, String reason) {
-            release();
```

Comment:
This worried me slightly but it looks fine because we catch the Exceptions so always call checkMailFinished.